Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -794,11 +796,19 @@ extension TaskHandler: ChannelDuplexHandler {
         assert(head.version == HTTPVersion(major: 1, minor: 1),
                "Sending a request in HTTP version \(head.version) which is unsupported by the above `if`")
 
+        self.expectedBodyLength = head.headers[canonicalForm: "content-length"].first.flatMap { Int($0) }
```

Comment:
this needs to error if there's more than one CL header

Reply:
shouldn't `validateHeaders` do that?

Reply:
I can add a return from `validate` with body length, as an alternative

Reply:
I'm happy if we have a test that proves that we error if the user sends two `content-length`s. In this code, I think we should just add an

```
assert(head.headers[canonicalForm: "content-length"].count <= 1)
```

to be sure.

Reply:
it's even better, we _remove_ all content length headers and insert them ourselves :) except one case, I'll add a test for that (and an assert).

Reply:
@artemredkin very nice! However if the user sets two CL headers, shouldn't we error? Maybe it's fine because we're (now) checking the body so the user should get an error anyway if we pick the "wrong" one :P. Just mean it could be confusing if the user does something like this

```
var headers = HTTPHeaders(["content-length": 17])
let request = ...(headers)
[...] // lots of code

func setBody(_ body: String, request: inout Request) {
    request.headers.add("content-length": "\(body.utf8.count)")
    request.body = ... .stream { writer in write.write(... body ...) }
}

setBody("hello", &request)
self.myHTTPClient.execute(request)
```

and now you'll get a `bodySizeMismatchError` but the code (in the `setBody` function) looks correct. So I just felt we _could_ be nicer to the user and diagnose the actual issue (which is that there are two CL headers present) instead of a follow up one (because we picked the first body header).

Said that I'm happy if we do sanitisation + checking of the actually sent body size.

Reply:
That will happen only if the stream has no length is not specified. In all other cases we actually ignore the user-provided headers, you think we should fail on those?

Reply:
@artemredkin wait, why are we looking for the first CL header then if this code is only run if the user specified _no_ CL headers?

Reply:
this code is run always, if the content-length was not specified, then `expectedLength` will be `nil` and final check will not be executed

Reply:
alright, let's just add the assertion&test (if not present for all the possible cases) in case somebody modifies the sanitisation code.

Reply:
done!